packages/components/pie-radio-group/test/visual/pie-radio-group.spec.ts

```diff
@@ -1,11 +1,77 @@
 import { test } from '@sand4rt/experimental-ct-web';
 import percySnapshot from '@percy/playwright';
+import {
+    type PropObject, type WebComponentPropValues, type WebComponentTestInput,
+} from '@justeattakeaway/pie-webc-testing/src/helpers/defs.ts';
+import {
+    getAllPropCombinations, splitCombinationsByPropertyValue,
+} from '@justeattakeaway/pie-webc-testing/src/helpers/get-all-prop-combos.ts';
+import { createTestWebComponent } from '@justeattakeaway/pie-webc-testing/src/helpers/rendering.ts';
+import { WebComponentTestWrapper } from '@justeattakeaway/pie-webc-testing/src/helpers/components/web-component-test-wrapper/WebComponentTestWrapper.ts';
+import { percyWidths } from '@justeattakeaway/pie-webc-testing/src/percy/breakpoints.ts';
+import { PieRadio } from '@justeattakeaway/pie-radio';
+import { PieFormLabel } from '@justeattakeaway/pie-form-label';
 import { PieRadioGroup } from '../../src/index.ts';
 
-test.describe('PieRadioGroup - Visual tests`', () => {
-    test('should display the PieRadioGroup component successfully', async ({ page, mount }) => {
-        await mount(PieRadioGroup);
+const props: PropObject = {
+    isInline: [true, false],
+    disabled: [true, false],
+    hasGroupLabel: [true, false],
+};
 
-        await percySnapshot(page, 'PieRadioGroup - Visual Test');
-    });
+const renderTestPieRadioGroup = (propVals: WebComponentPropValues) => {
+    let attributes = '';
+
+    if (propVals.isInline) attributes += ` isInline="${propVals.isInline}"`;
+    if (propVals.disabled) attributes += ` disabled="${propVals.disabled}"`;
+
+    return `
+    <pie-radio-group ${attributes}>
+        ${propVals.hasGroupLabel ? '<pie-form-label slot="label">Group Label</pie-form-label>' : ''}
+        <pie-radio value="radio-one">radio 1</pie-radio>
+        <pie-radio value="radio-two" checked>radio 2</pie-radio>
+        <pie-radio value="radio-three">radio 3</pie-radio>
+    </pie-radio-group>`;
+};
+
+const componentPropsMatrix: WebComponentPropValues[] = getAllPropCombinations(props);
+const componentPropsMatrixByInlineState: Record<string, WebComponentPropValues[]> = splitCombinationsByPropertyValue(componentPropsMatrix, 'isInline');
+const componentVariants: string[] = Object.keys(componentPropsMatrixByInlineState);
```

Comment:
You can remove these types if you like, because they can all be inferred by the methods
```suggestion
const componentPropsMatrix = getAllPropCombinations(props);
const componentPropsMatrixByInlineState = splitCombinationsByPropertyValue(componentPropsMatrix, 'isInline');
const componentVariants = Object.keys(componentPropsMatrixByInlineState);
```